x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/DefaultChunkingSettings.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.chunking;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.inference.ChunkingSettings;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public class DefaultChunkingSettings extends ChunkingSettings {
```

Comment:
Do we need a `DefaultChunkingSettings` class when the defaults can be represented by a instance of `WordBoundaryChunkingSettings`

```
public static final WordBoundaryChunkingSettings DEFAULT_SETTINGS = new WordBoundaryChunkingSettings(x, y);
```

Reply:
Those are all valid points and there are convincing arguments for both sides. I will answer the questions with respect to the conventions Elasticsearch uses.

1. If we have a notion of default without explicitly describing the settings and values then if the default changes (the code changes) it will change the behaviour for all existing configurations. In some situations this may be desirable but if we consider the upgrade scenario a user may be surprised to find that chunking has changed and that change is only manifested in code not in the configuration they can inspect. 

2. Default options are there to help the user by lightening the cognitive load when creating an inference endpoint, they should be persisted with the config so that they are visible to users. There are many examples of this - text embedding models default the similarity measure. If it is not set the default measure still appears in the config. In general it is better to surface the defaults so the user can learn about those settings. 

3.  True but there are other ways to distinguish between the user using the default and creating a config that is exactly the same as the default, for example a flag could be stored in the config and included in the metrics but not surfaced to the user. 

A good rule I learned from Elasticsearch is to follow the _principle of least surprise_, explicit defaults follow that principle.

Reply:
Makes sense. I think the best option then to ensure that the feature is clear to the user is to store a `WordBoundaryChunkingSettings` with the default values as part of the model configuration and fully remove the `DefaultChunkingSettings` object. Thanks for the suggestions on this.